### PR TITLE
Fix the output file for pip-compile with an explicit setup.py as source file

### LINKED ISSUE #730: pip-compile generates setup.txt by given explicitly setup.py in a source file
There is a possible wrong behavior on a command `pip-compile setup.py`, because it generates `setup.txt` file instead of `requirements.txt`.

##### Environment Versions

1. OS Type: Any (tested on MacOSX)
1. Python version: 3.7.x
1. pip version:  19.1
1. pip-tools version: 3.3.2

##### Steps to replicate

1. Prepare a `setup.py` file:
```
cat << EOF > setup.py
from setuptools import setup
setup(install_requires=[])
EOF
```
2. Run: `pip-compile setup.py`

##### Expected result

```
#
# This file is autogenerated by pip-compile
# To update, run:
#
#    pip-compile --output-file requirements.txt setup.py
#
```

##### Actual result

```
#
# This file is autogenerated by pip-compile
# To update, run:
#
#    pip-compile --output-file setup.txt setup.py
#
```

### PR DESCRIPTION
**Pull request by atugushev**
The command `pip-compile setup.py` generates a `setup.txt`, but
it should be a requirements.txt file.

Fixes #730.

**Changelog-friendly one-liner**: Fix the output file for pip-compile with an explicit setup.py as source file

##### Contributor checklist

- [X] Provided the tests for the changes.
- [X] Requested a review from another contributor.
- [X] Gave a clear one-line description in the PR (that the maintainers can add to CHANGELOG.md on release).
- [x] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

**Comment by atugushev** [comment]
Since a branch `src_files[0] == '-'` was touched i took responsibility to test that case.

File: piptools/scripts/compile.py [hunk 1]
```diff
@@ -78,15 +79,15 @@ def cli(verbose, quiet, dry_run, pre, rebuild, find_links, index_url, extra_inde
...
         else:
             raise click.BadParameter(("If you do not specify an input file, "
                                       "the default is {} or setup.py").format(DEFAULT_REQUIREMENTS_FILE))
 
-    if len(src_files) == 1 and src_files[0] == '-':
-        if not output_file:
+    if len(src_files) == 1:
+        if src_files[0] == '-' and not output_file:
             raise click.BadParameter('--output-file is required if input is from stdin')
+        elif src_files[0] == 'setup.py':
```
Comment (blueyed) [inline piptools/scripts/compile.py]: `src_files == ('setup.py',)` ?

Comment (atugushev) [reply piptools/scripts/compile.py]: I like that. No need for `len(src_files) == 1` then.

Comment (atugushev) [reply piptools/scripts/compile.py]: @blueyed i've fixed that block, see the latest commit please. Is it okay?

**Comment by atugushev** [comment]
Hello @blueyed. Gentle ping. Is everything okay in the PR. Could you approve or request any changes if you'd like.

**Comment by atugushev** [comment]
I've resolved conflicts, rebased and squashed commits. I'll merge this as soon as tests are passed.

@blueyed, thanks for your help!

**Comment by atugushev** [comment]
`pip-tools v3.5.0` is released

**Pull request merged**